### PR TITLE
Sc 3855/looker run query and download as file

### PR DESCRIPTION
**Pull request by Shamwayne**

### CODE REVIEW

File: looker_blueprints/run_query_and_download.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+EXIT_CODE_QUERY_ERROR = 203
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--base-url', dest='base_url', required=True)
+    parser.add_argument('--client-id', dest='client_id', required=True)
+    parser.add_argument('--client-secret', dest='client_secret', required=True)
+    parser.add_argument('--query-id', dest='query_id', required=False)
```
Comment (smjohnsonShipyard) [inline looker_blueprints/run_query_and_download.py]: query-id should be required. Also file_type, dest_file_name, dest_file_folder should be arguments here.

**Review by smjohnsonShipyard — CHANGES_REQUESTED** [review]

File: looker_blueprints/run_query_and_download.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                        required=True)
+    args = parser.parse_args()
+    return args
+
+
+def run_query_and_download(sdk, query_id, file_format):
+    try:
+        # Options are csv, json, json_detail, txt, html, md, xlsx, sql (raw query), png, jpg
+        response = sdk.run_query(query_id=query_id, result_format=file_format)
+        print("query {new_query.id} created successfully")
```
Comment (ericelsken) [inline looker_blueprints/run_query_and_download.py]: Looks like `new_query` doesn't exist and this isn't an f-string.

File: looker_blueprints/run_query_and_download.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--base-url', dest='base_url', required=True)
+    parser.add_argument('--client-id', dest='client_id', required=True)
+    parser.add_argument('--client-secret', dest='client_secret', required=True)
+    parser.add_argument('--query-id', dest='query_id', required=True)
+    parser.add_argument('--dest-file-name', dest='dest_file_name', required=True)
+    parser.add_argument('--dest-folder-name', dest='dest_folder_name', required=False)
+    parser.add_argument('--file-type', dest='file_type',
+                        choices=['json', 'json_detail','csv', 'html', 'xlsx', 'md', 'sql' ,'png', 'jpg'],
```
Comment (ericelsken) [inline looker_blueprints/run_query_and_download.py:23]: Formatting is off.

File: looker_blueprints/helpers.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        self.base_url = kw_args.pop('base_url')
+        self.client_id = kw_args.pop("client_id")
+        self.client_secret = kw_args.pop("client_secret")
+        super().__init__(*args, **kw_args)
+
+    def read_config(self):
+        config = super().read_config()
+        config['base_url'] = self.base_url
+        config["client_id"] = self.client_id
+        config["client_secret"] = self.client_secret
```
Comment (ericelsken) [inline looker_blueprints/helpers.py]: Make the use of single and double quotes consistent.

**Review by ericelsken — CHANGES_REQUESTED** [review]

**Pull request closed without merging**